modules/animation/src/animationcontroller.cpp

```diff
@@ -258,6 +258,50 @@ void AnimationController::stop() {
     eval(currentTime_, Seconds(0));
 }
 
+void AnimationController::nextKeyframe() {
+    auto times = animation_->getAllTimes();
+    auto it = std::upper_bound(times.begin(), times.end(), getCurrentTime());
+    if (it != times.end()) {
+        eval(getCurrentTime(), *it);
+    }
+}
+
+void AnimationController::prevKeyframe() {
+    auto times = animation_->getAllTimes();
+    auto it = std::lower_bound(times.begin(), times.end(), getCurrentTime());
+    if (it != times.begin()) {
+        eval(getCurrentTime(), *std::prev(it));
+    }
+}
+
+void AnimationController::nextControlKeyframe() {
+    std::vector<Seconds> times;
```

Comment:
Same here add a function to animation for finding the next time. Avoid allocation any vectors.